### ISSUE TITLE
Strange 2D layout

### ISSUE THREAD

**Issue by doctorn** [homotopy-io/homotopy-rs#426]
I discovered this layout when playing around with my [change actions](https://arxiv.org/abs/1902.05465?utm_source=feedburner&utm_medium=feed&utm_campaign=Feed%3A+arxiv%2FQSXk+%28ExcitingAds%21+cs+updates+on+arXiv.org%29) workspace.

<img width="319" alt="Screenshot 2022-03-14 at 14 46 25" src="https://user-images.githubusercontent.com/6421560/158196929-4ffc6d9c-d389-41c1-933c-1728b8f9145e.png">

[filename_todo.hom.zip](https://github.com/homotopy-io/homotopy-rs/files/8245875/filename_todo.hom.zip)

cc @calintat

**Comment by doctorn** [comment]
When the unit is at the same level as the action, it works fine.

<img width="400" alt="Screenshot 2022-03-14 at 14 52 58" src="https://user-images.githubusercontent.com/6421560/158198443-f3742ba4-c565-4ec0-bfbd-61428531cde1.png">

Then moving it, we get this issue.

<img width="321" alt="Screenshot 2022-03-14 at 14 52 45" src="https://user-images.githubusercontent.com/6421560/158198308-66404161-6130-47e8-a4ec-d354d7072ad0.png">

**Comment by doctorn** [comment]
Adding more instances of the action and the unit and performing similar homotopies, the results become increasingly wild.

<img width="640" alt="Screenshot 2022-03-14 at 14 53 32" src="https://user-images.githubusercontent.com/6421560/158198562-e586eacb-1638-4446-b14e-814a753615a7.png">
<img width="292" alt="Screenshot 2022-03-14 at 14 53 51" src="https://user-images.githubusercontent.com/6421560/158198586-2682de71-d5af-4d27-ba7f-29a31bf158cf.png">
<img width="437" alt="Screenshot 2022-03-14 at 14 54 02" src="https://user-images.githubusercontent.com/6421560/158198618-b6535b5a-fc09-4a6a-b921-dcc1cd67844c.png">

**Comment by calintat** [comment]
Oh dear, that last picture looks really messed up. I think it might be the same problem as #427 (i.e. an overflow subtraction), did you run this in debug mode?

**Comment by calintat** [comment]
I tested your workspace on a debug build and it panics with "attempt to subtract with overflow" so it's the same issue as #427. It's quite annoying that overflow subtractions are allowed in normal builds.

**Comment by calintat** [comment]
This is a minimal workspace to reproduce this: [overflow_subtraction.hom.zip](https://github.com/homotopy-io/homotopy-rs/files/8251979/overflow_subtraction.hom.zip)

<img width="236" alt="Screenshot 2022-03-15 at 09 46 47" src="https://user-images.githubusercontent.com/16837293/158351147-73ac7cbc-4d76-4265-bc49-487bcdfaa601.png">

**Comment by doctorn** [comment]
Is this just going to be a case of using `.saturating_sub` instead of `-`?

**Comment by calintat** [comment]
Not quite. I just changed how we calculate the width. I had assumed the width would be the same regardless of the alignment but that's no longer the case after the straight wires change. This part of the code is going to be rewritten as part of #419.
